express-routes/server.ts

```diff
@@ -0,0 +1,29 @@
+import express from 'express';
+
+const app = express();
+
+app.use((req, res, next) => {
+  console.log('request path', req.path);
+  console.log('request method', req.method);
+  console.log('The date is', new Date());
+  next();
+});
+
+app.listen(8080, () => {
+  console.log('Express server listening on port 8080');
+});
+
+app.get('/', (req, res, next) => {
+  res.send('get request to root');
+  next();
+});
+
+app.get('/notes', (req, res, next) => {
+  res.send('get to /notes');
+  next();
+});
+
+app.post('/notes/123', (req, res, next) => {
+  res.send('post to /notes/123');
+  next();
+});
```

Comment:
So `next()` works a little bit differently inside of routes rather than middleware. Generally we'll call `next()` at the end of our middleware to make sure to call any other following middlewares, but since we've already sent a response back to the user, we don't need to call `next`. In fact if we do, it could lead to additional errors, because it could try to change the `req` object that we already sent back after we sent it.

In a few exercises from now, you're going to learn about how to handle errors inside of a route, and that's when `next()` comes into play. So for now, it's not necessary to call `next()` inside of a route after you already sent a response back to the client.